File: rehovot-ads-packages/android/src/main/java/com/rehovottech/levelplay/RewardedManager.kt
package com.rehovottech.levelplay

// Rewarded concerns stay isolated to keep reward logic easy to audit later.
class RewardedManager(
    private val logger: Logger,
) {
    private var ready: Boolean = false

    public fun load(options: RewardedRequest) {
        logger.d("TODO: Load Rewarded")
        logger.d("Rewarded options: $options")
        ready = false
    }

    public fun show(options: RewardedRequest): RewardResultPayload {
        logger.d("TODO: Show Rewarded")
        logger.d("Rewarded options: $options")
        return RewardResultPayload(
            success = false,
            rewardGranted = false,
            message = "Unity LevelPlay SDK is not implemented yet.",
            placementId = options.placementId,
        )
    }

    public fun isReady(): Boolean {
        return ready
    }

    public fun destroy() {
        logger.d("TODO: Destroy Rewarded")
        ready = false
    }
}

File: rehovot-ads-packages/ios/Plugin/RewardedManager.swift
import Foundation

// Rewarded responsibilities remain separate so reward flow logic stays easy to audit.
final class RewardedManager {
    private let logger: Logger
    private var ready: Bool = false

    init(logger: Logger) {
        self.logger = logger
    }

    func load(_ options: RewardedRequest) {
        logger.debug("TODO: Load Rewarded")
        logger.debug("Rewarded options: \(options)")
        ready = false
    }

    func show(_ options: RewardedRequest) -> RewardResultPayload {
        logger.debug("TODO: Show Rewarded")
        logger.debug("Rewarded options: \(options)")
        return RewardResultPayload(
            success: false,
            rewardGranted: false,
            message: "Unity LevelPlay SDK is not implemented yet.",
            placementId: options.placementId,
            rewardAmount: options.rewardAmount,
            rewardCurrency: options.rewardCurrency
        )
    }

    func isReady() -> Bool {
        return ready
    }

    func destroy() {
        logger.debug("TODO: Destroy Rewarded")
        ready = false
    }
}
